.github/workflows/MergeTests.yml

```diff
@@ -0,0 +1,338 @@
+name: Merge Unit Tests
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build_for_testing:
+    runs-on: macos-latest
+    name: Build App
+    env:
+      DEVELOPER_DIR: /Applications/Xcode_14.2.0.app/Contents/Developer
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Install pip modules
+      run: |
+         python3 -m pip install virtualenv
+
+    - name: Install gems
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 2.7
+        bundler-cache: true
+
+    - name: Install npm
+      uses: bahmutov/npm-install@v1.6.0
+
+    - name: Restore Content Blocker Cache
+      id: restore-content-blocker-cache
+      uses: actions/cache/restore@v3
+      with:
+        path: |
+          content-blocker-lib-ios/
+          shavar-prod-lists/
+          Client/Assets/AllFramesAtDocument*.js
+          Client/Assets/MainFrameAtDocument*.js
+          Client/Assets/*.js.LICENSE.txt
+        key: ${{ runner.os }}-content-blocker-cache-${{ hashFiles('content_blocker_commit_sha.txt') }}
+
+    - name: Run content blocker scripts
+      if: steps.restore-content-blocker-cache.outputs.cache-hit != 'true'
+      run: ./content_blocker_update.sh
+
+    - name: Save Content Blocker Cache
+      if: steps.restore-spm-cache.outputs.cache-hit != 'true'
+      uses: actions/cache/save@v3
+      with:
+        path: |
+          content-blocker-lib-ios/
+          shavar-prod-lists/
+          Client/Assets/AllFramesAtDocument*.js
+          Client/Assets/MainFrameAtDocument*.js
+          Client/Assets/*.js.LICENSE.txt
+        key: ${{ runner.os }}-content-blocker-cache-${{ hashFiles('content_blocker_commit_sha.txt') }}
+
+    - name: Fix-up private Github URLs
+      run: find . -type f \( -name '*.pbxproj' -o -name 'Package.swift' -o -name 'Package.resolved' \) -exec sed -i '' "s/https:\/\/github.com\/${GITHUB_REPOSITORY_OWNER}/https:\/\/${GITHUB_ACCESS_TOKEN}@github.com\/${GITHUB_REPOSITORY_OWNER}/g" {} \;
+      env:
+        GITHUB_ACCESS_TOKEN: ${{ secrets.IOS_CORE_TOKEN }}
+
+    - name: Create empty Staging xcconfig
+      run: touch Client/Configuration/Staging.xcconfig
+
+    - name: Restore SPM Cache
+      id: restore-spm-cache
+      uses: actions/cache/restore@v3
+      with:
+        path: SourcePackages/
+        key: ${{ runner.os }}-spm-cache-${{ hashFiles('Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved') }}
+
+    - name: Build for testing
+      run: bundle exec fastlane build_and_export_derived_data_path
+
+    - name: Save derived data path into a txt file artifact
```

Comment:
Isn't the fastlane action `build_and_export_derived_data_path` executed above already storing the derived data path to this file?

Reply:
The fast lane action creates the file containing the derived data path.
In order to be able to exchange this path between steps and jobs (due to Github security reasons) I need to store it into a artefact and retrieve it back